main.js

```diff
@@ -52,60 +52,114 @@ const DEFAULT_PAGE_SIZE = 64; // Page size for retrieving documents from the cus
-
-    const migrator = new MigrationClient({
-      sourceKey: options.source,
-      targetKey: options.target,
-      defaultPageSize: DEFAULT_PAGE_SIZE,
-      maxParallelism: options.parallelism,
+    collectionList.push({
+      collectionName,
+      indexName,
     });
+  } else {
+    if (options.index) {
+      console.log(
+        "Custom index specified, but cannot be used when running with multiple collections.",
+      );
+    }
+    console.log(
+      "No collection specified. Gathering a list of all Collections in the source DB...",
+    );
+
+    const collectionNames = await migrator.listSourceCollections();
+
+    console.log(`Found ${collectionNames.length} collections:`);
+    console.log(collectionNames);
+    for (const collectionName of collectionNames) {
+      console.log(`  - ${collectionName}`);
+
+      collectionList.push({
+        collectionName,
+        indexName: "_migration_index_for_" + collectionName,
+      });
+    }
+  }
+
+  if (options.validate) {
+    for (const collection of collectionList) {
+      const { collectionName, indexName } = collection;
+
+      const validationOptions = {
+        ...options,
+        collection: collectionName,
+        index: indexName,
+      };
 
+      console.log(`Validating complete migration of Collection("${collectionName}")`);
+      await validate(validationOptions);
+    }
+  } else {
+    console.log();
     console.log("Initializing UDFs...");
     await migrator.initializeSourceFunctions();
+    console.log();
 
-    console.log(`Initializing Collection '${collectionName}'...`);
-    const initialized = await migrator.initializeCollection({
-      collectionName,
-      indexName,
-    });
-    if (!initialized) {
+    console.log(`Initializing Collections...`);
+    let collectionsInitialized = true;
+    for (const collection of collectionList) {
+      const { collectionName, indexName } = collection;
+
+      const initialized = await migrator.initializeCollection({
+        collectionName,
+        indexName,
+      });
+
+      collectionsInitialized = collectionsInitialized && initialized;
+    }
+    if (!collectionsInitialized) {
       throw new Error("Initialization error; can't continue");
     }
-
-    console.log(
-      `BEGIN synchronizing events in collection '${collectionName}' at ${new Date().toISOString()}...`,
-    );
+    console.log();
 
     let iterations = ITERATIONS;
     let startTime = options.timestamp;
 
     do {
-      await migrator.migrateCollection({
-        collectionName,
-        indexName,
-        startTime,
-        duration: DURATION,
-      });
+      console.log(
+        `BEGIN synchronizing events at ${new Date().toISOString()}...`,
+      );
+      console.log();
 
-      startTime += DURATION * 60 * 1000 * 1000; //increase the start time by 'DURATION' amount of minutes at every iteration
+      for (const collection of collectionList) {
```

Comment:
Question: if users have cross-collection references in their docs, would we want/need to fold all the events from all collections into a single queue of events and migrate that way?

Reply:
As long as the collection exists, there is no problem creating a reference to a document that does not exist. One can still have dangling references in their database regardless of this tool.